src/apply_diff.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{StringDiffOp,StringDiffOpKind};
+pub(crate) fn remove(start: usize, stop: usize, s: &str) -> String {
+	assert!(stop > start);
+	let mut result = String::from("");
+	for (i, c) in s.chars().enumerate() {
+		if start > i || stop < i + 1 {
+			result.push(c);
+		}
+	}
+	result
+}
+
+pub fn apply_diff(s: &str, diffs: Vec<StringDiffOp>) -> String {
+	let mut new_string: String = s.into();
+
+	for i in diffs.iter() {
+		match i.kind {
+			StringDiffOpKind::Delete(_x) => {
+				new_string = remove(i.index, i.index + 1, &new_string);
+			}
+			StringDiffOpKind::Insert(_x) => {
+				new_string.push(_x);
+			}
+			StringDiffOpKind::Substitute(_x, _y) => {
+				new_string.replace_range((i.index)..(i.index + 1), &_y.to_string())
+			}
+			StringDiffOpKind::Transpose(_x, _y) => {
+				panic!("apply_diff does not currently support the transpose operation yet")
+			}
+		}
+	}
+	new_string.clone()
+}
+
+
+#[cfg(test)]
+mod dcode_tests {
+
+	fn vec_compare(va: &Vec<StringDiffOp>, vb: &Vec<StringDiffOp>) -> bool {
+		(va.len() == vb.len()) &&  // zip stops at the shortest
+        va.iter()
+        .zip(vb)
+        .all(|(a,b)| a == b)
+	}
```

Comment:
`vec_compare()` doesn't get called inside this file (it gets called in `hamming.rs` and `levenshtein.rs`), so you can remove it here